src/modules/git_state.rs

```diff
@@ -12,37 +13,43 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let mut module = context.new_module("git_state");
     let config: GitStateConfig = GitStateConfig::try_load(module.config);
 
-    module.set_style(config.style);
-    module.get_prefix().set_value("(");
-    module.get_suffix().set_value(") ");
-
     let repo = context.get_repo().ok()?;
     let repo_root = repo.root.as_ref()?;
     let repo_state = repo.state?;
 
-    let state_description = get_state_description(repo_state, repo_root, config);
-
-    let label = match &state_description {
-        StateDescription::Label(label) => label,
-        StateDescription::LabelAndProgress(label, _) => label,
-        StateDescription::Clean => {
+    let state_description = get_state_description(repo_state, repo_root, &config)?;
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "state" => Some(Ok(state_description.label)),
```

Comment:
What do you think about making `label` format strings, so that users can write e.g. `[REBASING](bold green)` to change the styles?

```suggestion
            .map_meta(|variable, _| match variable {
                "state" => Some(state_description.label),
```

Reply:
I think that's a good change! Should I update the docs to reflect this somehow?

Reply:
Yes, go ahead :+1: